clients/src/main/java/org/apache/kafka/common/metrics/ClientTelemetryReceiver.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.common.metrics;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+@InterfaceStability.Evolving
+public interface ClientTelemetryReceiver {
```

Comment:
Is that idea that an implementation of the `ClientTelemetryReceiver` would decouple the implementation of `exportMetrics` from a call out to the upstream metrics system (e.g. Datadog)?

Reply:
I guess my understanding was that the broker-side telemetry plugin would do little more than take what it received as input and push to the desired metrics repository. In which case a naive plugin implementation would block the network thread of the broker for all users while it's pushing the data out to, say, Datadog.

Reply:
@kirktrue Since we have not figured out what to be done in the plugin and how to configure it in CDC environments etc.. for now hoping that we take care about those scenarios when we go to the next phase of the development where we are going to implement a test/example plugin. 

IMHO right thing todo here is to make an async call and respond back to the client whenever that call is finished. Or else we can leave that implementation to the plugin itself.

Reply:
Works for me 😄